### PR TITLE
[4.2] Add missing properties

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This simple PR adds missing properties to various classes in our libraries to avoid deprecated warning missing properties in PHP 8.2. These missing properties are detected automatically by rector [CompleteDynamicPropertiesRector](https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#completedynamicpropertiesrector) rule.

### Testing Instructions
1. Code review should be enough
2. Or apply patch, installing extension and make sure it is still working OK.

### Actual result BEFORE applying this Pull Request
Deprecated warnings when installing package extensions on PHP 8.2 and in some other places.

### Expected result AFTER applying this Pull Request
No warnings anymore.

### Link to documentations
- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
@joomdonation PHPCS complains about the new variable with the name beginning with an underscore here:

```
FILE: /********/src/libraries/src/Installer/Manifest.php
----------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
----------------------------------------------------------------------
 31 | WARNING | Property name "$_errors" should not be prefixed with
    |         | an underscore to indicate visibility
----------------------------------------------------------------------
```
You can solve that by adding file "libraries/src/Installer/Manifest.php" to the list of allowed exceptions in the ruleset in this section: https://github.com/joomla/joomla-cms/blob/4.2-dev/ruleset.xml#L32 like it has been already done for file "libraries/src/Client/FtpClient.php", which is full of such variable names.

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on f1c41ffa689f5689550a8e528d84c251926a108a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39691">issues.joomla.org/tracker/joomla-cms/39691</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on f1c41ffa689f5689550a8e528d84c251926a108a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39691">issues.joomla.org/tracker/joomla-cms/39691</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39691">issues.joomla.org/tracker/joomla-cms/39691</a>.</sub>

File: ruleset.xml [hunk 1]
```diff
@@ -94,6 +94,7 @@
         <exclude-pattern type="relative">libraries/src/Encrypt/Totp\.php</exclude-pattern>
         <exclude-pattern type="relative">libraries/src/Form/Field/CaptchaField\.php</exclude-pattern>
         <exclude-pattern type="relative">libraries/src/Input/Json\.php</exclude-pattern>
+        <exclude-pattern type="relative">libraries/src/Installer/Manifest\.php</exclude-pattern>
```
Comment (laoneo) [inline ruleset.xml:97]: Why is this needed?

Comment (joomdonation) [reply ruleset.xml:97]: It is needed to fix phpcs error as @richard67 explained here https://github.com/joomla/joomla-cms/pull/39691#issuecomment-1399446089

Comment (laoneo) [reply ruleset.xml:97]: Ok, thanks!

**Comment by laoneo** [comment]
The error variable needs to be deprecated then in 4.3 as it is probably a leftover in conjunction with CMSObject and not used anymore.

**Comment by fancyFranci** [comment]
Thanks for fixing the warnings!
